core-primitives/test/src/mock/onchain_mock.rs

```diff
@@ -160,22 +132,50 @@ impl EnclaveMetricsOCallApi for OnchainMock {
 	}
 }
 
-// We cannot implement EnclaveOnChainOCallApi specifically here, because OnchainMock already
-// implements `GetStorageVerified`. And all implementers of `EnclaveOnChainOCallApi` automatically
-// implement GetStorageVerified too (-> see `core-primitives/storage-verified/src/lib.rs`),
-// so it results in duplicate implementations.
-// impl EnclaveOnChainOCallApi for OnchainMock {
-// 	fn send_to_parentchain(&self, _extrinsics: Vec<OpaqueExtrinsic>) -> SgxResult<()> {
-// 		Ok(())
-// 	}
-//
-// 	fn worker_request<V: Encode + Decode>(
-// 		&self,
-// 		_req: Vec<WorkerRequest>,
-// 	) -> SgxResult<Vec<WorkerResponse<V>>> {
-// 		Ok(Vec::new())
-// 	}
-// }
+impl EnclaveOnChainOCallApi for OnchainMock {
```

Comment:
This has pained me before (as you can read in the comment), so I'm glad we resolved this now 😄

Reply:
Haha, I thought at some point that this is a good pattern. I can see why we want to remove the generic trait implementations. However, what speaks against having it as a separate trait, such that you can have the trait bounds as minimal as possible?

Because, as we can see here in the mock, we only need the two methods related to `GetStorageVerified`.

Reply:
`get_storage_verified` itself needs the function `worker_request` (https://github.com/integritee-network/worker/blob/bh-clean-up/enclave-runtime/src/ocall/on_chain_ocall.rs#L103-L105), so separating them completely doesn't seem an option to me. Or do you mean something else?

Reply:
Well, builds upon the function `worker_request`, which is the lowest abstraction layer; hence you could make that trait having the `EncalveOcallApi` as trait bound, just as it was before, or even remove the trait bound.

All generic things with `<T: GetStorageVerified>`  just need to know that `T` knows how to get the storage verified, the low-level abstraction behind is irrelevant. By merging `GetStorageVerified` into `EnclaveOcallApi`, this separation between low-level and high-level is removed.

Reply:
Btw. I don't have a strong opinion here, but your mock implementation shows exactly what I am saying above. You need to implement unnecessarily two functions because they are in the same trait.

Reply:
If you can live with this implementation for now I'd love to leave as is. The problem is, I don't think it's correct to implement `get_storage_verfied` for the `OcallApi`. But if we don't implement it for the OcallApi, we'd need to change a lot of inputs.. 

I believe we should clean this one up once we implement the parentchain-block-subscriber (or light client, don't remember which one which will implement the node connection).

What do you think? Should I create an appropriate issue?

Reply:
No, this is ok. I think is ok to implement `GetStorageVerified` for the `OcallApi`.

With my following statement, I am only 90% sure:

Your thinking comes form object-oriented programming, where you want to encapsulate things. In rust, the paradigm is different. You just extend structs, satisfying a certain trait with another trait. Rust removes this inherent need for encapsulation. (It might make sense in some contexts though).

Reply:
Yes, I agree, it's a downgrade in the Trait bound perspective. Sorry for that.

Reply:
Okidoki, I will be silent now. Sorry for the trouble. :)